File: api/src/main/java/com/fundmate/api/service/AccountService.java
package com.fundmate.api.service;

import com.fundmate.api.dto.request.AccountRequest;
import com.fundmate.api.dto.response.AccountResponse;

import java.util.List;

public interface AccountService {
    AccountResponse createAccount(AccountRequest accountRequest);
    List<AccountResponse> getAccounts();
    AccountResponse getAccountById(Long accountId);
    AccountResponse updateAccount(Long accountId, AccountRequest accountRequest);
    void deleteAccount(Long accountId);
}

File: frontend/src/services/account.service.js
import api from "./api";

export const accountService = {
  async getAccounts() {
    const response = await api.get("/accounts");
    return response.data;
  },

  async createAccount(accountData) {
    const response = await api.post("/accounts", accountData);
    return response.data;
  },

  async updateAccount(id, accountData) {
    const response = await api.put(`/accounts/${id}`, accountData);
    return response.data;
  },

  async deleteAccount(id) {
    const response = await api.delete(`/accounts/${id}`);
    return response.data;
  },
};
